src/docs/codelabs/extended-flex-widget.md

```diff
@@ -0,0 +1,364 @@
+---
+title: "Codelab: basic Flutter layout concepts"
+description: "A codelab that teaches basic Flutter layout concepts through DartPad2 code examples."
+toc: true
+--- 
+{{site.alert.important}}
+  This codelab covers basic Flutter layout concepts using a code editor called DartPad2. An understanding of [Dart](https://www.dart.dev) or some exposure to object-oriented programming is recommended. 
+{{site.alert.end}}
+{{site.alert.important}}
+  Currently, DartPad2 is in the experimental stage and being tested with Google Chrome. A PC (Microsoft, Mac, or Linux) with Google Chrome installed is required for this codelab. If you encounter any difficulties while using DartPad2, please create a [DartPad issue](https://github.com/dart-lang/dart-pad/issues/new) and label the issue "platform-web."
+{{site.alert.end}} 
+{{site.alert.note}}
```

Comment:
Has this come up?

Reply:
The only issue I encountered is the one where I exceeded the number of times I could refresh the page, and the DartPad2s didn't reload. However, I think I'll leave this because the note is entered twice on Andrew's codelab. https://flutter.dev/docs/codelabs/layout-basics